src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp

```diff
@@ -288,88 +439,246 @@ class npc_stalagg : public CreatureScript
+                    return;
+                if (!isFeignDeath && me->IsInCombat() && !me->GetHomePosition().IsInDist(me, OVERLOAD_DISTANCE))
+                {
+                    if (!isOverloading)
+                    {
+                        isOverloading = true;
+                        caster->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                        caster->Talk(EMOTE_TESLA_LINK_BREAKS, CHAT_MSG_RAID_BOSS_EMOTE, 50000.0f, NULL);
+                        me->RemoveAura(SPELL_STALAGG_CHAIN_VISUAL);
+                    }
+                    if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM))
+                    {
+                        caster->CastStop(SPELL_TESLA_SHOCK);
+                        caster->CastSpell(target, SPELL_TESLA_SHOCK,true);
                     }
                 }
+                else if (isOverloading || refreshBeam)
+                {
+                    isOverloading = false;
+                    refreshBeam = false;
+                    caster->CastStop();
+                    caster->CastSpell(me, SPELL_STALAGG_CHAIN_VISUAL, true);
+                    caster->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                }
+            }
+
+            void UpdateAI(uint32 uiDiff) override
+            {
+                if (isFeignDeath)
+                    return;
+                if (!UpdateVictim())
+                    return;
+
+                if (powerSurgeTimer <= uiDiff)
+                {
+                    DoCast(me, SPELL_STALAGG_POWERSURGE);
+                    powerSurgeTimer = urand(15, 20) * IN_MILLISECONDS;
+                }
+                else
+                    powerSurgeTimer -= uiDiff;
 
-                magneticPullTimer = 20000;
+                DoMeleeAttackIfReady();
             }
-            else magneticPullTimer -= uiDiff;
 
-            if (powerSurgeTimer <= uiDiff)
+        private:
+            Creature* myCoil()
             {
-                DoCast(me, SPELL_POWERSURGE);
-                powerSurgeTimer = urand(15000, 20000);
-            } else powerSurgeTimer -= uiDiff;
+                Creature* coil = NULL;
+                if (_myCoil)
+                    coil = ObjectAccessor::GetCreature(*me, _myCoil);
+                if (!coil)
+                    if ((coil = me->FindNearestCreature(NPC_TESLA, 1000.0f, true)))
+                    {
+                        _myCoil = coil->GetGUID();
+                        coil->SetReactState(REACT_PASSIVE);
+                    }
+                return coil;
+            }
 
-            DoMeleeAttackIfReady();
-        }
+            GameObject* myCoilGO()
+            {
+                GameObject* coil = NULL;
+                if (_myCoilGO)
+                    coil = ObjectAccessor::GetGameObject(*me, _myCoilGO);
+                if (!coil)
+                    if ((coil = me->FindNearestGameObject(GO_CONS_NOX_TESLA_STALAGG, 1000.0f)))
```

Comment:
This triggers a warning about assignment inside if - to silence it you must explicitly compare it by appending != nullptr